src/Dropzone/assets/src/controller.ts

```diff
@@ -56,38 +65,60 @@ export default class extends Controller {
         this.inputTarget.value = '';
         this.inputTarget.style.display = 'block';
         this.placeholderTarget.style.display = 'block';
+        this.previewTarget.innerHTML = '';
         this.previewTarget.style.display = 'none';
-        this.previewImageTarget.style.display = 'none';
-        this.previewImageTarget.style.backgroundImage = 'none';
-        this.previewFilenameTarget.textContent = '';
 
         this.dispatchEvent('clear');
     }
 
     onInputChange(event: any) {
-        const file = event.target.files[0];
-        if (typeof file === 'undefined') {
+        const files = event.target.files;
+        if (files.length === 0) {
+            this.previewClearButtonTarget.style.display = 'none';
             return;
         }
 
         // Hide the input and placeholder
         this.inputTarget.style.display = 'none';
         this.placeholderTarget.style.display = 'none';
 
-        // Show the filename in preview
-        this.previewFilenameTarget.textContent = file.name;
-        this.previewTarget.style.display = 'flex';
+        // Clear previous previews
+        this.previewTarget.innerHTML = '';
+
+        for (const file of files) {
+            // Create a container for each file preview
+            const filePreviewContainer = document.createElement('div');
+            filePreviewContainer.classList.add('dropzone-preview-file');
+
+            // Create a filename preview element
+            const fileNameElement = document.createElement('span');
+            fileNameElement.textContent = file.name;
+            filePreviewContainer.appendChild(fileNameElement);
+
+            // Create an image preview element if the file is an image, else a default svg file icon
+            if (file.type) {
+                const imagePreviewElement = document.createElement('div');
+                if (file.type.indexOf('image') !== -1) {
+                    imagePreviewElement.classList.add('dropzone-preview-image');
+                    this._populateImagePreview(file, imagePreviewElement);
+                } else {
+                    imagePreviewElement.classList.add('dropzone-preview-svg');
+                }
```

Comment:
Why svg ?

Reply:
it is the svg "file" icon when file is not an image

Reply:
Ok so here I think you should not expose this as it's not the "semantic" here, and we don't want user to feel they need an svg.

What about dropzone-preview-file or dropzone-no-preview .

You get the idea ?